applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -164,11 +158,80 @@ void TwoFluidNavierStokes<TElementData>::CalculateLocalSystem(
                     ComputeGaussPointEnrichmentContributions(data, Vtot, Htot, Kee_tot, rhs_ee_tot);
                 }
 
+                Matrix int_shape_function, int_shape_function_enr_neg, int_shape_function_enr_pos;
+                GeometryType::ShapeFunctionsGradientsType int_shape_derivatives;
+                Vector int_gauss_pts_weights;
+                std::vector< array_1d<double,3> > int_normals_neg;
+
+                if (rCurrentProcessInfo[SURFACE_TENSION] || rCurrentProcessInfo[MOMENTUM_CORRECTION]){
+                    ComputeSplitInterface(
+                        data,
+                        int_shape_function,
+                        int_shape_function_enr_pos,
+                        int_shape_function_enr_neg,
+                        int_shape_derivatives,
+                        int_gauss_pts_weights,
+                        int_normals_neg,
+                        p_modified_sh_func);
+                }
+
+                if (rCurrentProcessInfo[MOMENTUM_CORRECTION]){
+                    MatrixType lhs_acc_correction = ZeroMatrix(NumNodes*NumNodes,NumNodes*NumNodes);
```

Comment:
```suggestion
                    BoundedMatrix<double, LocalSize, LocalSize> lhs_acc_correction = ZeroMatrix(LocalSize,LocalSize);
```
This can be a stack array. Also note that you can use `LocalSize`.

Reply:
As far as I see, using a stack array is possible but would make the code less readable. I prefer not to use it here.

Reply:
This has a huge impact in performance. Stack arrays must be always used if possible. Note that there is no change in code readability as it only changes the declaration of the variable from `MatrixType a` to `BoundedMatrix<double,LocalSize,LocalSize b`.

Reply:
Please ignore my stupid comment. Corrected.